### DISCUSSION TITLE
Flask-SocketIO and socketserver UDP port 514

### DISCUSSION THREAD

**Discussion by schussda** [miguelgrinberg/Flask-SocketIO#1792] [Q&A]
I am using Flask-SocketIO and i am starting a socketserver in Windows10

import socketserver
handler = socketserver.UDPServer(("", 514), handle_func)

i can not run this code anymore when using Flask-SocketIO, is it possible to change the port for Flask-SocketIO?
I can not change the port because i am listening to syslog events on UDP port 514.

  File "C:\Program Files\Python39\lib\socketserver.py", line 466, in server_bind
    self.socket.bind(self.server_address)
OSError: [WinError 10048] Only one usage of each socket address (protocol/network address/port) is normally permitted

**Comment by miguelgrinberg** [comment]
I don't understand the problem. Flask-SocketIO doesn't use UDP nor port 514.

**Reply by schussda** [reply]
Thanks so much for your reply. You are right. I figured out my code is executed twice. For any reason it is restarted when following line is executed:

from flask_socketio import SocketIO, emit
from flask import request, Flask
app = Flask("Server")
socketio = SocketIO(app, logger=True, engineio_logger=True)
**socketio.run(app, host="10.60.2.138", port=5000, debug=True)**

Server initialized for eventlet.
Restarting with watchdog (windowsapi)

Do you know the reason for that?

**Reply by schussda** [reply]
I found out it can be disabled via use_reloader=False

After disabling it. the error is gone...

Is it bad not reloading?

**Reply by schussda** [reply]
I got another question 

I use a broadcast and send json data to all clients from within a thread (started via **threading.Thread**)
After 20sec a timeout occurs. Then the connection is re established and the json data is send on connect (no broadcast) which works fine

@socketio.event
def connect(auth):
    print(f"Client '{request.environ['REMOTE_ADDR']}' connected.")
    emit("new_config", json.dumps(MyServer.CONFIG_DATA, indent=4))

Log with broadcast problem:
emitting event "new_config" to all [/]
y647BaVz9C9LzIoYAAAC: Sending packet MESSAGE data 2 ...
**y647BaVz9C9LzIoYAAAC: Unexpected error "timed out", closing connection**
10.60.2.138 - - [16/Feb/2022 07:45:14] "GET /socket.io/?EIO=4&transport=websocket&sid=y647BaVz9C9LzIoYAAAC HTTP/1.1" 200 0 44.994483
W0Z9-q0i6D1wlRLIAAAE: Sending packet OPEN data {'sid': 'W0Z9-q0i6D1wlRLIAAAE', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}
10.60.2.138 - - [16/Feb/2022 07:45:15] "GET /socket.io/?EIO=4&transport=polling&t=Ny1HokO HTTP/1.1" 200 278 0.000998
W0Z9-q0i6D1wlRLIAAAE: Received packet MESSAGE data 0
Client '10.60.2.138' connected.
emitting event "new_config" to eqnrwQ0I3Np3wJctAAAF [/]
W0Z9-q0i6D1wlRLIAAAE: Sending packet MESSAGE data 2 ...
W0Z9-q0i6D1wlRLIAAAE: Sending packet MESSAGE data 0{"sid":"eqnrwQ0I3Np3wJctAAAF"}
10.60.2.138 - - [16/Feb/2022 07:45:15] "POST /socket.io/?EIO=4&transport=polling&t=Ny1HokV&sid=W0Z9-q0i6D1wlRLIAAAE HTTP/1.1" 200 221 0.003990
(15448) accepted ('10.60.2.138', 49194)
W0Z9-q0i6D1wlRLIAAAE: Received request to upgrade to websocket
10.60.2.138 - - [16/Feb/2022 07:45:15] "GET /socket.io/?EIO=4&transport=polling&t=Ny1HokW&sid=W0Z9-q0i6D1wlRLIAAAE HTTP/1.1" 200 181 0.001001
W0Z9-q0i6D1wlRLIAAAE: Upgrade to websocket successful
emitting event "new_config" to all [/]



on server in a (json is about 15KB)
self.WebServer.emit("new_config", json.dumps(self.CONFIG_DATA, indent=4))

on client:
socketio.on('new_config', function(json_config) {
    console.log("new config");
    current_json = json_config;
    update(json_config);
});

**Reply by miguelgrinberg** [reply]
I can't really tell you, but this is probably related to bad usage of eventlet in terms of blocking calls. If you don't understand how eventlet works I recommend that you switch to the `threading` mode, which is more compatible with standard Python code.

**Reply by schussda** [reply]
thanks for your reply, i found a solution
=>i do not not send a broadcast in the thread where socketserver  serve_forever function is running
=> just changing a variable of my class and sending a broadcast in another thread that is running anyway

i absolutely do not know how eventlet works, thats my first time using web sockets
I watched your youtube video tutorial about it and changed our server/client from http request to websockets and it works pretty well now.
